app/models/concerns/wp_cache.rb

```diff
@@ -48,11 +56,31 @@ def create_or_update(wp_type, wp_id, preview = false)
     # the `update_wp_cache` instance method.
     # Removes records with unknown IDs.
     #
-    def create_or_update_all
-      wp_json = get_from_wp_api self.to_s.underscore.pluralize
+    # TODO (dunyakirkali) clean up
+    def create_or_update_all_paginated
+      page = 0
+      ids = []
+      max_page = (ENV['MAX_PAGE'].to_i == 0 ? 999 : ENV['MAX_PAGE'].to_i)
+      while page < max_page do
+        puts " page #{page}"
+        wp_json = get_from_wp_api(wp_type, page)
+        break if wp_json.empty?
+        ids << wp_json.map do |json|
+          wp_id = json['ID']
+          where(wp_id: wp_id).first_or_initialize.update_wp_cache(json)
+          wp_id
+        end
+        page = page + 1
+      end
+      where('wp_id NOT IN (?)', ids).destroy_all unless ids.empty?
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.